src/controllers/userController.js

```diff
@@ -0,0 +1,90 @@
+'use strict';
+
+const userService = require('../services/userService');
+
+async function createUser(req, res) {
+  const { name } = req.body;
+
+  if (!name) {
+    return res.status(400).send('Name and  are required');
+  }
+
+  try {
+    const newUser = await userService.createUser({ name });
+
+    res.status(201).json(newUser);
+  } catch (error) {
+    res.status(500).send('Error creating user');
+  }
+}
+
+async function getAllUsers(req, res) {
+  try {
+    const users = await userService.getAllUsers(req.query);
+
+    res.json(users);
+  } catch (error) {
+    res.status(500).send('Error fetching users');
+  }
+}
+
+async function getUser(req, res) {
+  try {
+    const users = await userService.getAllUsers({
+      id: parseInt(req.params.id),
+    });
+    const user = users[0];
+
+    if (!user) {
+      return res.status(404).send('User not found');
+    }
+
+    res.json(user);
+  } catch (error) {
+    res.status(500).send('Error fetching user');
+  }
+}
+
+async function updateUser(req, res) {
+  const { name } = req.body;
+
+  if (!name) {
+    return res.status(400).send('Name and  are required');
```

Comment:
Name and what?

- This test is the same as seen before, maybe you could store it in a validation function?